File: sdks/dotnet/src/Croniq.Runner.Sdk/Protocol/RegisterJobRequest.cs
using System.Text.Json.Serialization;

namespace Croniq.Runner.Sdk.Protocol;

internal sealed record RegisterJobRequest(
    [property: JsonPropertyName("job_key")] string JobKey,
    [property: JsonPropertyName("schedule")] string Schedule,
    [property: JsonPropertyName("timezone"), JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)] string? Timezone,
    [property: JsonPropertyName("timeout"), JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)] string? Timeout,
    [property: JsonPropertyName("runner_id"), JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)] string? RunnerId,
    [property: JsonPropertyName("capabilities")] IReadOnlyList<string> Capabilities,
    [property: JsonPropertyName("description"), JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)] string? Description);

File: sdks/java/core/src/main/java/io/croniq/runner/protocol/RegisterJobRequest.java
package io.croniq.runner.protocol;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import java.util.List;

/** POST /v1/jobs/register request body. Used by self-registration in PR-4. */
@JsonInclude(JsonInclude.Include.NON_NULL)
public record RegisterJobRequest(
        @JsonProperty("job_key") String jobKey,
        @JsonProperty("schedule") String schedule,
        @JsonProperty("timezone") String timezone,
        @JsonProperty("timeout") String timeout,
        @JsonProperty("runner_id") String runnerId,
        @JsonProperty("capabilities") List<String> capabilities,
        @JsonProperty("description") String description) {}
